### PR TITLE
amd media - device support

### PR DESCRIPTION
**Pull request by rrawther**
Added parameter to specify device_id for amd_media decoder

### CODE REVIEW

File: amd_openvx_extensions/amd_media/decoder.cpp [hunk 1]
```diff
@@ -163,6 +165,7 @@ static inline int num_hw_devices() {
 
 static int hw_decoder_init(AVCodecContext *ctx, const enum AVHWDeviceType type, AVBufferRef *hw_device_ctx, int hw_device_id)
 {
+    //printf("hw_decoder_init called for device_id:%d\n", hw_device_id);
```
Comment (asalmanp) [inline amd_openvx_extensions/amd_media/decoder.cpp:168]: please remove this comment, if it is not needed.

File: amd_openvx_extensions/amd_media/decoder.cpp [hunk 2]
```diff
@@ -358,20 +363,29 @@ CLoomIoMediaDecoder::~CLoomIoMediaDecoder()
...
+vx_status CLoomIoMediaDecoder::SetEnableUserBufferGPUMode(vx_bool bEnable) {
     m_enableUserBufferGPU = bEnable;
     return VX_SUCCESS;
 }
 
+vx_status CLoomIoMediaDecoder::SetDeviceId(vx_int32 device_id_mask) {
+    // use default of device_id_mask is -1
+    bool use_default = (device_id_mask == -1);
+    for (int mediaIndex = 0; mediaIndex < mediaCount; mediaIndex++) {
+        hwDeviceID[mediaIndex] = use_default? -1 : (device_id_mask & 0xF);
```
Comment (asalmanp) [inline amd_openvx_extensions/amd_media/decoder.cpp:382]: minor formatting: add a space before question mark

File: amd_openvx_extensions/amd_media/decoder.cpp [hunk 3]
```diff
@@ -520,6 +534,7 @@ vx_status CLoomIoMediaDecoder::Initialize()
             return -1;
         if (useVaapi[mediaIndex]) {
             codecContext->get_format  = get_hw_format;
+            int dev_id = (hwDeviceID[mediaIndex] < 0) ? mediaIndex : hwDeviceID[mediaIndex];
```
Comment (asalmanp) [inline amd_openvx_extensions/amd_media/decoder.cpp:537]: where do you use the dev_id? did you mean to pass it to the hw_decoder_init function?

File: amd_openvx_extensions/amd_media/decoder.cpp (hunk 1 above)
Comment (rrawther) [reply amd_openvx_extensions/amd_media/decoder.cpp:168]: done

File: amd_openvx_extensions/amd_media/decoder.cpp (hunk 3 above)
Comment (rrawther) [reply amd_openvx_extensions/amd_media/decoder.cpp:537]: Yes. Good catch

**Review by asalmanp — APPROVED** [review]

**Pull request merged**